### DISCUSSION TITLE
Does this Support .NET MAUI (.NET 8). Android is not working with .NET 8 when we use MediaGallery.CapturePhotoAsync().

### DISCUSSION THREAD

**Discussion by thanigaiharan** [dimonovdd/Xamarin.MediaGallery#127] [General]
Android is not working with .NET 8 when we use MediaGallery.CapturePhotoAsync() and MediaGallery.PickAsync().

**Comment by dimonovdd** [comment]
What are the errors?

**Comment by thanigaiharan** [comment]
@dimonovdd  It shows no error but when we execute  var photo = await MediaGallery.CapturePhotoAsync(); it opens the camera viewer and when take photo and click On "Ok" in preview page, it is not returning anything, No exceptions.  and not even the error. 

Break point not hitting on next line.
  if (photo == null)
        return photoPath;
        
        Using .NET SDK 8.0.204

**Reply by beeradmoore** [reply]
Can you make a small repository with base required stuff to have a picker appear and get pick photos? (Including all your permissions and stuff)

**Reply by thanigaiharan** [reply]
@beeradmoore  Added the sample code here https://github.com/thanigaiharan/Mediagallery_Issue_Test.

**Reply by dimonovdd** [reply]
the repository is empty

**Reply by thanigaiharan** [reply]
https://github.com/thanigaiharan/Mediagallery_Issue_Test

Added as zip file

**Reply by beeradmoore** [reply]
@thanigaiharan , I downloaded your zip but it failed to run. `OnCounterClicked` is meant to be `OnPhotoClicked`. In future when you create reproduction repositories can you just put the code in there rather than a zip. I was out today when I tried to look into it but I couldn't view the code to see if something obvious was off because it was in a zip that I can't access on my phone. Plus can you 100% make sure that code runs when you commit it. It slows my little bit of free time that I have wanting to help you.

When I did get it to run I wasn't able to get any useful tests and I now have to head off for the night. There is something odd happening where I can't have a breakpoint after it to see if it's null, but it is also not throwing any exceptions. I am unsure if this is a library issue or a MAUI debugging issue. When I can dig in again I'll see if I can get more info out of it.

**Reply by thanigaiharan** [reply]
@beeradmoore Sorry for the issue am uploading code to Git repository for the first time , same behaviour is happening for me as well.  when I put break point for null check of photo it does not hit also not throwing any exceptions. 
It is happening in release mode as well.  This MediaGallery.CapturePhotoAsync() code opens the camera viewer and does nothing after we pressed "Ok" in preview page. 

This is happening in Release mode as well.

**Reply by thanigaiharan** [reply]
@beeradmoore we used this nugget in our  legacy(Xamairn Native) application which was used world wide by 100K users. we want to use this in our MAUI application as well as it very stable in legacy application . Please let me know if this issue can be resolved . Thanks !

**Comment by thanigaiharan** [comment]
@dimonovdd @beeradmoore Is this issue resolvable? Do we need to look for other options? Please advice.

**Reply by dimonovdd** [reply]
I can't open your zip file for the next two weeks. Upload the code to the repository without archiving

@beeradmoore by the way, I plan to abandon Xamarin support in June.

**Reply by beeradmoore** [reply]
@thanigaiharan I will try find time to look this weekend.


@dimonovdd sounds good to me!

**Reply by beeradmoore** [reply]
@thanigaiharan , I did not get time to test on the weekend, still awaiting time to give this another shot.

**Reply by beeradmoore** [reply]
@thanigaiharan , 

I was not able to get this to work on my macOS machine. Talking about unrelated problems in the MAUI repo the last reply was that my Android SDK/tools don't seem to be working correctly. But in saying that it could also be because I run Android 14 on a Pixel 2 XL and it doesn't like that. Could also be faulty USB cables (that I have swapped out). It could have been a USB hub (that I stopped using). 

Putting it onto my Windows machine that I don't use for Android development I couldn't even get it to compile. Issues with restoring the nuget. Kept saying things that it shouldn't like how net8.0-android is not compatible with net8.0-windows, even when I removed that and net8.0-macos as build targets. 

Something ain't right, but I still have no idea if its this project, or both of my dev machines, or my Android dev phone.

**Comment by thanigaiharan** [comment]
@beeradmoore Below is the sample code , I created new MAUI Project and added the below code in button Event. 

    private async void OnPhotoClicked(object sender, EventArgs e)
    {
        await this.CheckAndRequestCameraPermission();

        try
        {
            if (MediaGallery.CheckCapturePhotoSupport())
            {
                var photo = await MediaGallery.CapturePhotoAsync();

                if (photo != null)
                {

                }
            }
        }
        catch (Exception ex)
        {

        }

    }

    private async Task CheckAndRequestCameraPermission()
    {
        await Permissions.RequestAsync<Permissions.Camera>();
    }

******************************** Below is the Android Manifest Permission **********

        <uses-permission android:name="android.permission.ACCESS_MEDIA_LOCATION" />
	<uses-permission android:name="android.permission.CAPTURE_VIDEO_OUTPUT" />
	<uses-permission android:name="android.permission.CAPTURE_SECURE_VIDEO_OUTPUT" />
	<uses-permission android:name="android.permission.CAMERA" />
	<uses-permission android:name="android.permission.CAPTURE_AUDIO_OUTPUT" />

**Comment by thanigaiharan** [comment]
@beeradmoore Please let me know if we have fix for this issue. Thanks
